xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/main/java/com/xpn/xwiki/tool/backup/DataMojo.java

```diff
@@ -71,4 +75,40 @@ public void executeInternal() throws MojoExecutionException
 
         this.extensionHelper.install(this.includes, installRequest, "wiki:xwiki", null);
     }
+
+    @Override
+    protected void after() throws MojoExecutionException
+    {
+        consumeIndexingQueue();
+        super.after();
+    }
+
+    /**
+     * Waits for the document indexing tasks to be completed before continuing.
+     *
+     * @throws MojoExecutionException in case of error when resolving the components or when stopping the thread
+     */
+    private void consumeIndexingQueue() throws MojoExecutionException
+    {
+        ComponentManager componentManager =
+            (ComponentManager) this.oldCoreHelper.getXWikiContext().get(ComponentManager.class.getName());
+        try {
+            // Manually start the consumer threads as it is not automatically started in the packaging setup.
+            // This is also convenient as we can be sure that the tasks are all queued before being consumed.
+            TaskManager taskManager = componentManager.getInstance(TaskManager.class);
+            if (taskManager instanceof DefaultTasksManager) {
+                ((DefaultTasksManager) taskManager).startThread();
```

Comment:
Why is this needed ? I see that this is done automatically in TaskApplicationReadyListener when receiving ApplicationReadyEvent and ApplicationReadyEvent seems to be sent in OldCoreHelper#create.

Reply:
`org.xwiki.index.internal.TaskApplicationReadyListener#initialize` is called but the context provider returns null. 
`org.xwiki.index.internal.TaskApplicationReadyListener#onEvent` is never called, which not be right.

Reply:
> org.xwiki.index.internal.TaskApplicationReadyListener#initialize is called but the context provider returns null.

This is expected yes, same as in a standard XWiki instance.

>  org.xwiki.index.internal.TaskApplicationReadyListener#onEvent is never called, which not be right.

That's really not expected according to https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/main/java/org/xwiki/tool/utils/OldCoreHelper.java#L152.

Reply:
Alright, debugging.

Reply:
> That's really not expected according to [https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/main/java/org/xwiki/tool/utils/OldCoreHelper.java#L152](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/main/java/org/xwiki/tool/utils/OldCoreHelper.java?rgh-link-date=2022-03-21T15%3A40%3A31Z#L152).

Actually this is ApplicationStartedEvent sorry and indeed we don't seems to send ApplicationReadyEvent.